api/src/main/java/org/apache/iceberg/io/CloseableIterable.java

```diff
@@ -130,6 +130,20 @@ public O next() {
       private ConcatCloseableIterator(Iterable<CloseableIterable<E>> inputs) {
         this.iterables = inputs.iterator();
         this.currentIterable = iterables.next();
+        // Exhaust the starting empty iterables.
+        while (Iterables.isEmpty(currentIterable)) {
```

Comment:
Would it be simpler to simply call `hasNext` in `next` method? like so
```
@Override
      public E next() {
        if (hasNext()) {
          return currentIterator.next();
        } else {
          throw new NoSuchElementException();
        }
      }
```

My understanding is we can keep calling `hasNext` as many times as we want until we consume that element from the Iterator.

Reply:
+1, Make sense.

Reply:
+1

Reply:
Thanks for the suggestion, let me change it.